File: Java/1039.java
import java.util.Scanner;

public class Main {
    /*
    Author: Yuki
    GitHub: https://github.com/Yuki-14544869/
    Blog:   https://yuki-14544869.github.io/
    */
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int T = in.nextInt();
        while(T-- > 0) {
            StringBuffer input = new StringBuffer(in.next());
            int ans = 0;
            int len = input.length();
            for(int i=0; i<len; ++i) {
                for(char ch='A'; ch<='C'; ++ch) {
                    StringBuffer tmp = new StringBuffer(input);
                    tmp = tmp.insert(i, ch);
                    ans = Math.max(ans, cal(tmp));
                }
            }
            System.out.println(ans);
        }
        in.close();
    }
    public static int cal(StringBuffer string) {
        int len = string.length();
        if(len<=0)
            return 0;
        StringBuffer buffer = new StringBuffer("");
        int l = 0;
        string.append("*");
        for(int i=1; i<len+1; ++i) {
            if(string.charAt(i) != string.charAt(i-1)) {
                if(l == i-1) {
                    buffer.append(string.charAt(i-1));
                }
                l = i;
            }
        }
        if (buffer.length() == len) {
            return 0;
        }
        return len-buffer.length()+cal(buffer);
    }
}

File: C++/1039.cpp
/*
    Author: Yuki
    GitHub: https://github.com/Yuki-14544869/
    Blog:   https://yuki-14544869.github.io/
*/
#include <bits/stdc++.h>

using namespace std;
const int INF=0x3f3f3f3f;

int cal(string p) {
    int len = p.size();
    if(p.empty())
        return 0;
    string t = "";
    int l = 0;
    p += "*";
    for(int i=1; p[i]; i++) {
        if(p[i] != p[i-1]) {
            if(l == i-1) {
                t += p[i-1];
            }
            l = i;
        }
    }
    if(t.size() == len)
        return 0;
    return len-t.size()+cal(t);
}

int main() {
#ifndef ONLINE_JUDGE
    freopen("in.txt", "r", stdin);
#endif
    int t;
    cin >> t;
    while(t--) {
        string s;
        string insert[3] = {"A","B","C"};
        cin >> s;
        int ans=0;
        for(int i=0; s[i]; i++) {
            for(int j=0; j<3; ++j) {
                string tmp = s;
                tmp.insert(i, insert[j]);
                ans = max(ans, cal(tmp));
            }
        }
        cout << ans << endl;
    }
    return 0;
}